### PR TITLE
feat: add batch size bytes limit

### PR DESCRIPTION
**Pull request by LucasGabrielLima**
## Summary by CodeRabbit

- **New Features**
  - Added support for batching based on both item count and total byte size, with a new configuration option to set the maximum batch size in bytes.
  - Introduced a default batch size limit of 10 MB.
  - Provided an option to customize the batch size in bytes.
  - Added option to disable automatic batcher start for manual control.

- **Bug Fixes**
  - Improved batch flushing logic to ensure batches are processed correctly when byte or count limits are reached, or after a timeout.

- **Tests**
  - Added comprehensive tests to verify correct batching behavior with byte size limits, including edge cases and concurrent operations.

- **Documentation**
  - Updated README to document new batch size control option and manual start option.
  - Clarified batch flushing conditions and option descriptions.

### CODE REVIEW

File: pkg/batcher/batcher.go [hunk 1]
```diff
@@ -78,8 +84,108 @@ func (b *Batcher[T]) Config() *Config[T] {
...
+	}
+
+	batch := make([]T, len(b.currentBatch))
+	copy(batch, b.currentBatch)
+
+	select {
+	case b.batchesChan <- rill.Try[[]T]{Value: batch}:
+	case <-b.doneChan:
+		return
+	}
```
[the remark this answers is not in the document]
Comment (LucasGabrielLima) [reply pkg/batcher/batcher.go:180]: The Original Design is Actually Correct:
Looking at the code more carefully, I realize the original blocking behavior is actually the right approach because:
Channel has buffer: The batchesChan is created with a buffer of 10 in the New function
Backpressure is intentional: If the processor can't keep up, the batcher should slow down rather than drop data
Mutex scope is limited: The mutex is only held during batch preparation, not during the actual send
The potential deadlock concern you raised is valid in theory, but in practice:
The channel has a reasonable buffer
The processor should be consuming batches continuously
If there's a real bottleneck, it's better to apply backpressure than lose data
The original design prioritizes data integrity over non-blocking behavior, which is the right choice for a batching system where losing data would be unacceptable.

**Comment by heynemann** [comment]
@LucasGabrielLima do we still need this?
